src/plugins/es_ui_shared/static/forms/hook_form_lib/hooks/use_field.ts

```diff
@@ -0,0 +1,378 @@
+    errorDisplayDelay = form.options.errorDisplayDelay,
+    serializer = (value: unknown) => value,
+    deSerializer = (value: unknown) => value,
+  } = config;
+
+  const [value, setStateValue] = useState(
+    typeof defaultValue === 'function' ? deSerializer(defaultValue()) : deSerializer(defaultValue)
+  );
+  const [errors, setErrors] = useState<ValidationError[]>([]);
+  const [isPristine, setPristine] = useState(true);
+  const [isValidating, setValidating] = useState(false);
+  const [isUpdating, setIsUpdating] = useState(false);
+  const validateCounter = useRef(0);
+  const debounceTimeout = useRef<NodeJS.Timeout | null>(null);
+
+  // -- HELPERS
+  // ----------------------------------
+
+  /**
+   * Filter an array of errors with specific validation type on them
+   *
+   * @param _errors The array of errors to filter
+   * @param validationType The validation type to filter out
+   */
+  const filterErrors = (
+    _errors: ValidationError[],
+    validationTypeToFilterOut: string | string[] = VALIDATION_TYPES.FIELD
+  ): ValidationError[] => {
+    const validationTypeToArray = Array.isArray(validationTypeToFilterOut)
+      ? (validationTypeToFilterOut as string[])
+      : ([validationTypeToFilterOut] as string[]);
+
+    return _errors.filter(error =>
+      validationTypeToArray.every(_type => error.validationType !== _type)
+    );
+  };
+
+  const runFormatters = (input: unknown): unknown => {
+    const isEmptyString = typeof input === 'string' && input.trim() === '';
+
+    if (isEmptyString) {
+      return input;
+    }
+    return formatters.reduce((output, formatter) => formatter(output), input);
+  };
+
+  const onValueChange = () => {
+    if (isPristine) {
+      setPristine(false);
+    }
+    setIsUpdating(true);
+
+    if (debounceTimeout.current) {
+      clearTimeout(debounceTimeout.current);
+    }
+
+    debounceTimeout.current = setTimeout(() => {
+      setIsUpdating(false);
+    }, errorDisplayDelay);
+  };
+
+  const validateSync = ({
+    formData,
+    value: valueToValidate,
+    validationTypeToValidate,
+  }: {
+    formData: any;
+    value: unknown;
+    validationTypeToValidate?: string;
+  }): ValidationError[] => {
+    const validationErrors: ValidationError[] = [];
+    let skip = false;
+
+    const runValidation = ({
+      validator,
+      exitOnFail,
+      type: validationType = VALIDATION_TYPES.FIELD,
+    }: ValidationConfig) => {
+      if (
+        skip ||
+        (typeof validationTypeToValidate !== 'undefined' &&
+          validationType !== validationTypeToValidate)
+      ) {
+        return;
+      }
+      let validationResult;
+
+      try {
+        validationResult = validator({
+          value: (valueToValidate as unknown) as string,
+          errors: validationErrors,
+          formData,
+          path,
+        });
+
+        if (validationResult && exitOnFail !== false) {
+          throw validationResult;
+        }
+      } catch (error) {
```

Comment:
This will silently swallow any errors emitted by the validator function (e.g. trying to an undefined method or accessing a property on null), making debugging validators difficult. You can verify this by using a validator like this:

```js
  const schema = {
    title: {
      validations: [{
        validator: ({ value }) => {
          if ((value).asdf('.')) {
            return {
              code: 'ERR_FORMAT_ERROR',
              message: `The title can't start with a dot (.)`,
            };
          }
        },
      }]
    }
  }
```

Reply:
Good point! This has been resolved by updating the `validateSync` and `validateAsync` from your comment above.